ksqldb-engine/src/main/java/io/confluent/ksql/internal/PullQueryExecutorMetrics.java

```diff
@@ -699,5 +773,31 @@ public String toString() {
           + ", routingNodeType=" + routingNodeType
           + '}';
     }
+
+    public String variantName() {
+      return sourceTypeName() + "-"
+          + planTypeName() + "-"
+          + routingNodeTypeName();
+    }
+
+    public String sourceTypeName() {
+      return getName(sourceType);
+    }
+
+    public String planTypeName() {
+      return getName(planType);
+    }
+
+    public String routingNodeTypeName() {
+      return getName(routingNodeType);
+    }
+
+    private String getName(final Enum<?> o) {
+      if (o == null) {
+        return "unknown";
```

Comment:
That's a good idea. I specifically avoided adding nullable references to the enum values as a way to get away from "unknown" because, I agree, that's a strictly worse way to do the same thing. I really wanted to capture that it's not that a specific attribute of the query/result can be unknown, but rather that we want to register metrics some times in error cases when we just don't know what kind of query was run, for example because we couldn't parse it.